pkg/registry/pod/rest.go

```diff
@@ -59,6 +59,14 @@ func (podStrategy) ResetBeforeCreate(obj runtime.Object) {
 	}
 }
 
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (podStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	newPod := obj.(*api.Pod)
+	oldPod := old.(*api.Pod)
+	newPod.Status = oldPod.Status
+	//FIXME: can Spec.Host be updated?
```

Comment:
Intent was for prepare to do things that involve stripping invalid user input based on the semantics of the call.  Status, things that are not intended to be allowed to change.  Then validate runs and should fail if the user provides something.

Currently ValidateObjectMetaUpdate is doing some checks (creationTimestamp, etc) that should be handled before PrepareForUpdate and Validation are called:

Create
1. Object meta is normalized
2. PrepareForCreate
3. ValidateCreate

Update
1. Object meta is normalized
2. PrepareForUpdate
3. ValidateUpdate

----- Original Message -----
> > @@ -59,6 +59,14 @@ func (podStrategy) ResetBeforeCreate(obj runtime.Object)
> > {
> >  	}
> >  }
> >  
> > +// PrepareForUpdate clears fields that are not allowed to be set by end
> > users on update.
> > +func (podStrategy) PrepareForUpdate(obj, old runtime.Object) {
> > +	newPod := obj.(*api.Pod)
> > +	oldPod := old.(*api.Pod)
> > +	newPod.Status = oldPod.Status
> > +	//FIXME: can Spec.Host be updated?
> 
> That's a good point.  @smarterclayton should we validate first?  Or should we
> allow Prepare to return errors?  Or should we intentionally copy Host over
> and let it fail in validation?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/5970/files#r27182480